js/src/forum/components/SsoGrid.js

```diff
@@ -0,0 +1,147 @@
+
+import Component from '../../common/Component';
+import Checkbox from '../../common/components/Checkbox';
+import icon from '../../common/helpers/icon';
+import ItemList from '../../common/utils/ItemList';
+
+/**
+ * The `SsoGrid` component displays a table of linked sso providers,
+ * allowing the user to link/unlink each one to their account.
+ *
+ */
+export default class SsoGrid extends Component {
+  init() {
+    /**
+     * Information about the available Sso methods.
+     *
+     * @type {Array}
+     */
+    this.providers = this.ssoProviders().toArray();
+
+    /**
+     * A map of Sso type-method combinations to the checkbox instances
+     * that represent them.
+     *
+     * @type {Object}
+     */
+    this.inputs = {};
+
+    // For each of the Sso type-method combinations, create and store a
+    // new checkbox component instance, which we will render in the view.
+    this.providers.forEach(provider => {
+      this.inputs[provider.name] = Checkbox.component({
```

Comment:
Same comment as with the Checkbox in `AuthPage.js`. It should be created inside of `view()` https://mithril.js.org/components.html#avoid-creating-component-instances-outside-views

Reply:
Done!